lib/src/lints/flutter/prefer_dedicated_media_query_method.dart

```diff
@@ -54,6 +53,29 @@ class PreferDedicatedMediaQueryMethod extends DartLintRule {
   ) {
     if (!context.pubspec.isFlutterProject) return;
 
+    context.registry.addPrefixedIdentifier((node) {
+      final initializer = node;
+
+      final typeName = initializer.prefix.staticType
+          ?.getDisplayString(withNullability: false);
+      if (typeName != 'MediaQueryData') return;
```

Comment:
We should probably use `TypeChecker` instead of `DartType.getDisplayString` here.
```dart
const mediaQueryDataChecker = TypeChecker.fromName(
  'MediaQueryData',
  packageName: 'flutter',
);

if (targetType == null ||
          !mediaQueryDataChecker.isExactlyType(targetType)) return;
```